src/components/Onboarding/restoreSecretKey/RestoreSecretKey.tsx

```diff
@@ -68,44 +64,43 @@ export const RestoreSecretKey = ({ goToStep }: { goToStep: (step: OnboardingStep
       icon={<KeyIcon width="24px" height="24px" stroke={colors.gray[450]} />}
       title="Insert Secret Key"
     >
-      <form onSubmit={handleSubmit(onSubmit)} style={{ width: "100%" }}>
-        <FormControl isInvalid={!!errors.secretKey}>
-          <FormLabel>Secret Key</FormLabel>
-          <Textarea
-            minHeight="130px"
-            data-testid="secret-key"
-            {...register("secretKey", {
-              required: "Secret key is required",
-              // taken from https://github.com/ecadlabs/taquito/blob/master/packages/taquito-signer/src/taquito-signer.ts#L95
-              onChange: event =>
-                setIsEncrypted(isEncryptedSecretKeyPrefix(event.target.value.trim())),
-            })}
-            placeholder="Your secret key"
-          />
-          {errors.secretKey && <FormErrorMessage>{errors.secretKey.message}</FormErrorMessage>}
-        </FormControl>
-
-        {isEncrypted && (
-          <FormControl marginTop="20px" isInvalid={!!errors.password}>
-            <FormLabel>Password</FormLabel>
-            <Input
-              data-testid="password"
-              {...register("password", {
-                validate: value => {
-                  if (!value.trim()) {
-                    return "Password is required";
-                  }
-                },
+      <FormProvider {...form}>
+        <form onSubmit={handleSubmit(onSubmit)} style={{ width: "100%" }}>
+          <FormControl isInvalid={!!errors.secretKey}>
+            <FormLabel>Secret Key</FormLabel>
+            <Textarea
+              minHeight="130px"
+              data-testid="secret-key"
+              {...register("secretKey", {
+                required: "Secret key is required",
+                // taken from https://github.com/ecadlabs/taquito/blob/master/packages/taquito-signer/src/taquito-signer.ts#L95
+                onChange: event =>
+                  setIsEncrypted(isEncryptedSecretKeyPrefix(event.target.value.trim())),
               })}
+              placeholder="Your secret key"
             />
-            {errors.password && <FormErrorMessage>{errors.password.message}</FormErrorMessage>}
+            {errors.secretKey && <FormErrorMessage>{errors.secretKey.message}</FormErrorMessage>}
           </FormControl>
-        )}
 
-        <Button width="100%" marginTop="32px" isDisabled={!isValid} size="lg" type="submit">
-          Continue
-        </Button>
-      </form>
+          {isEncrypted && (
+            <FormControl marginTop="20px" isInvalid={!!errors.password}>
+              <PasswordInput data-testid="password" inputName="password" minLength={0} />
```

Comment:
Just wanted to check the reasoning for `minLength={0}`

I think we have `minLength=8` in Umami 🤔 
However, for random secret keys it makes sense to me to allow different values